spec/requests/groups_controller_spec.rb

```diff
@@ -829,6 +829,101 @@ def expect_type_to_return_right_groups(type, expected_group_ids)
       end
     end
 
+    context "when user is a site moderator" do
+      before do
+        user.update!(moderator: true)
```

Comment:
It might be cleaner to use a `Fabricate(:moderator)` rather than updating a `user` to be a moderator. I'll leave it up to you to consider this as perhaps there's another reason for using the existing `user` instance.

Reply:
I may have been thinking "the user has now been promoted to moderator". But, you're right: there is a moderator already being Fabricated, so I'll use that.